kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/adapters/ALBookingRMAdapter.java

```diff
@@ -0,0 +1,522 @@
+            final ALRichMessageModel.ALPayloadModel payload = payloadList.get(position);
+
+            if (!TextUtils.isEmpty(payload.getHeaderImageUrl())) {
+                Glide.with(context).load(payload.getHeaderImageUrl()).into(viewHolder.productImage);
+                viewHolder.productImage.setVisibility(View.VISIBLE);
+                viewHolder.productImageOverlay.setVisibility(View.VISIBLE);
+                viewHolder.productPrice.setBackground(context.getResources().getDrawable(R.drawable.al_rich_messaging_price_border));
+            } else {
+                viewHolder.productImage.setVisibility(View.GONE);
+                viewHolder.productImageOverlay.setVisibility(GONE);
+                viewHolder.productPrice.setBackground(context.getResources().getDrawable(R.drawable.al_imageless_rich_message_price_border));
+            }
+
+            if (!TextUtils.isEmpty(payload.getOverlayText())) {
+                viewHolder.productPrice.setText(payload.getOverlayText());
+                viewHolder.productPrice.setVisibility(View.VISIBLE);
+            } else {
+                viewHolder.productPrice.setVisibility(View.GONE);
+            }
+
+            if (TextUtils.isEmpty(payload.getRating())) {
+                viewHolder.productNameSplitLayout.setVisibility(GONE);
+                if (!TextUtils.isEmpty(payload.getTitle())) {
+                    viewHolder.productNameSingleLine.setVisibility(View.VISIBLE);
+                    viewHolder.productNameSingleLine.setText(payload.getTitle());
+                } else {
+                    viewHolder.productNameSingleLine.setVisibility(View.GONE);
+                }
+            } else {
+                viewHolder.productNameSplitLayout.setVisibility(View.VISIBLE);
+                viewHolder.productNameSingleLine.setVisibility(View.GONE);
+                viewHolder.productName.setVisibility(View.VISIBLE);
+                viewHolder.productRating.setText(payload.getRating());
+
+                if (!TextUtils.isEmpty(payload.getTitle())) {
+                    viewHolder.productName.setText(payload.getTitle());
+                } else {
+                    viewHolder.productName.setText("");
+                }
+            }
+
+            if (!TextUtils.isEmpty(payload.getSubtitle())) {
+                viewHolder.productLocation.setVisibility(View.VISIBLE);
+                viewHolder.productLocation.setText(AlRichMessage.getHtmlText(payload.getSubtitle()));
+            } else {
+                viewHolder.productLocation.setVisibility(View.GONE);
+            }
+
+            if (!TextUtils.isEmpty(payload.getDescription())) {
+                viewHolder.productDescription.setVisibility(View.VISIBLE);
+                viewHolder.productDescription.setText(AlRichMessage.getHtmlText(payload.getDescription()));
+            } else {
+                viewHolder.productDescription.setVisibility(View.GONE);
+            }
+
+            for (int i = 0; i < 3; i++) {
```

Comment:
what's the reason for limiting to 3?

Reply:
in that case, we can replace 3 with a constant

Reply:
replaced with constant